config/config.go

```diff
@@ -22,15 +23,15 @@ const (
 
 // Config is the overall config for the framework, holding configurations for supported networks
 type Config struct {
-	Network            string                    `mapstructure:"network" yaml:"network"`
-	Logging            *LoggingConfig            `mapstructure:"logging" yaml:"logging"`
-	Networks           map[string]*NetworkConfig `mapstructure:"networks" yaml:"networks"`
-	Retry              *RetryConfig              `mapstructure:"retry" yaml:"retry"`
-	Apps               AppConfig                 `mapstructure:"apps" yaml:"apps"`
-	Kubernetes         KubernetesConfig          `mapstructure:"kubernetes" yaml:"kubernetes"`
-	KeepEnvironments   string                    `mapstructure:"keep_environments" yaml:"keep_environments"`
-	Prometheus         *PrometheusConfig         `mapstructure:"prometheus" yaml:"prometheus"`
-	Contracts          *ContractsConfig          `mapstructure:"contracts" yaml:"contracts"`
+	Network            string                   `mapstructure:"network" yaml:"network"`
```

Comment:
We want to be able to define which networks at runtime, like we do now with a single network, so I'd change this to `Networks []string`. This way we can leave `ethereum_geth,ethereum_geth` as the default and at runtime, set the ENV:
`NETWORKS=matic_mumbai,binance_testnet`
etc

This way we don't need to define networks in the multiple networks in tests like you've got with the unit test. Then in CI, we can run the multi-network suite with a matrix in GHA that runs the tests on all network combinations.